numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,29 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    # __new__ is overriden as a clean way to provide decorator-with-keywords
+    # syntax while allowing 'vectorize' to remain as the type of vectorized
+    # functions.
+    def __new__(cls, pyfunc=None, *args, **kwargs):
+
+        # For backwards compatibility, subclasses must not use the overloaded
+        # __new__, as vectorize.__new__ and subclass.__init__ will receive the
+        # same arguments.
+        if cls is not vectorize:
```

Comment:
Below, `vectorize_decorator` is created unnecessarily for the case that `pyfunc` is passed in, so that check could be done here. Now could we also solve the subclass isssue by just making this 
```
if pyfunc is not None:
    return object.__new__(cls)
```
At least, I'm hoping that any existing subclasses would not start to give different meaning to the first argument being `None`...